### PR TITLE
azurerm_public_ip resource: deprecated public_ip_address_allocation 

### PR DESCRIPTION
**Pull request by ajaguilar-viewnext**
Changed to allocation_method. Values are: Dynamic or Static. 
I also changed value in var.public_ip_address_allocation in order to use case-sensitive value (Dynamic as default) but I did not change variable name for backward compatibility.

### CODE REVIEW

**Comment by ajaguilar-viewnext** [comment]
Hi @JunyiYi. Sorry about that. The only commit I wanted to this PR is the first one.
Maybe I had a mistake when configuring my PR.

**Comment by JunyiYi** [comment]
@ajaguilar-viewnext I see, please update the PR so we could merge it.

File: main.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 provider "azurerm" {
-  version = ">= 1.1.0"
+  version = "= 1.21.0"
```
Comment (metacpp) [inline main.tf]: Can we use version >= `1.24.0` and < `2.0.0` ?
```suggestion
  version = "~> 1.2"
```

File: main.tf [hunk 2]
```diff
@@ -273,6 +274,32 @@ resource "azurerm_network_security_group" "vm" {
     destination_address_prefix = "*"
   }
 
+  security_rule {
```
Comment (metacpp) [inline main.tf]: @ajaguilar-viewnext can we revert the changes for `security-rule` ?

File: main.tf [hunk 3]
```diff
@@ -82,6 +82,7 @@ resource "azurerm_virtual_machine" "vm-linux" {
     enabled     = "${var.boot_diagnostics}"
     storage_uri = "${var.boot_diagnostics == "true" ? join(",", azurerm_storage_account.vm-sa.*.primary_blob_endpoint) : "" }"
   }
+
```
Comment (metacpp) [inline main.tf]: Can we remove this blank line ?
```suggestion
```

**Comment by metacpp** [comment]
@ajaguilar-viewnext it looks like that your change met some styling issue from terraform, can we use `terraform fmt` to reformat the changed files?

```
$ docker run ${IMAGE_NAME} rake build
INFO: Styling Terraform configurations...
rake aborted!
ERROR: Styling Terraform configurations failed!
main.tf
```

File: main.tf (hunk 2 above)
Comment (ajaguilar-viewnext) [reply main.tf]: Absolutely

**Comment by yupwei68** [comment]
Hi @ajaguilar-viewnext ,thanks for your contribution. For consideration of merge conflicts of your PR, we've recently release v3.0.0 which include your features. Sorry that I have to close this PR.
